Sources/WalletConnect/Engine/SessionEngine.swift

```diff
@@ -196,16 +185,19 @@ final class SessionEngine {
     
     func update(topic: String, accounts: Set<String>) throws {
         guard var session = sequencesStore.getSequence(forTopic: topic) else {
-            throw WalletConnectError.internal(.noSequenceForTopic)
+            throw WalletConnectError.noSessionMatchingTopic(topic)
+        }
+        guard session.isSettled else {
+            throw WalletConnectError.sessionNotSettled(topic)
+        }
+        guard session.selfIsController else {
+            throw WalletConnectError.unauthorizedNonControllerCall
         }
         for account in accounts {
             if !String.conformsToCAIP10(account) {
-                throw WalletConnectError.internal(.notApproved) // TODO: Use a suitable error cases
+                throw WalletConnectError.invalidCAIP10Account(account)
```

Comment:
if we use `Set<Account>` as `update()` param instead of `Set<String>` we won't have to check it.

Reply:
Maybe it's even better if the `Account` object is stored in session sequences instead of strings.